### ISSUE TITLE
Convert Moments/Strategies commenting mechanism from jQuery to React

### ISSUE THREAD

**Issue by julianguyen** [ifmeorg/ifme#966]
# Description 

Convert Moments/Strategies commenting mechanism from jQuery to React. This is part of our long term goal of getting rid of JS code on the Rails side.

The Notifications widget is a good example of how to use API calls in the client.

# Screenshots

![image](https://user-images.githubusercontent.com/3010728/45689500-fa673380-bb08-11e8-97bd-c52ed105a712.png)

# Do you want to be the assignee to work on this?

🚫

**Comment by wshatch** [comment]
Is this the stuff in `add/delete_comment.js`?

**Comment by julianguyen** [comment]
@wshatch Yep! So I recommend moving on that into client/widgets. You might have to also update things in the controller so that you can nicely use the API from the client.

**Comment by wshatch** [comment]
Gotcha, on it.

**Comment by julianguyen** [comment]
Hey @wshatch, are you actively working on this? If so can you assign yourself to this issue? If you're unable to do so please follow these [steps](https://github.com/ifmeorg/ifme/wiki/Join-Our-Slack).

**Comment by julianguyen** [comment]
Taking over this issue since there is a lot of complicated refactoring, I have a [WIP branch](https://github.com/ifmeorg/ifme/tree/comments-to-react). Almost done, writing more tests :)
